modules/ROOT/pages/values-and-types/property-structural-constructed.adoc

```diff
@@ -50,23 +50,55 @@ In Neo4j, all relationships have a direction.
 However, you can have the notion of undirected relationships at query time.
 ====
 
-[[composite-types]]
-== Composite types
+[[constructed-types]]
+== Constructed types
 
-The following data types are included in the composite types category: `List` and `Map`.
+The following data types are included in the constructed types category: `LIST` and `MAP`.
 
-* Composite types can be returned from Cypher queries.
-* Composite types can be used as xref::syntax/parameters.adoc[parameters].
-* Composite types cannot be stored as properties (with the exception of homogenous lists).
-* Composite types can be constructed with xref::syntax/expressions.adoc[Cypher literals].
+* Constructed types can be returned from Cypher queries.
+* Constructed types can be used as xref::syntax/parameters.adoc[parameters].
+* Constructed types cannot be stored as properties (with the exception of homogenous lists).
+* Constructed types can be constructed with xref::syntax/expressions.adoc[Cypher literals].
 
-The `List` data type can be either a homogenous collection of simple values, or a heterogeneous, ordered collection of values, each of which can have any property, structural or composite type.
+The `LIST` data type can be either a homogenous collection of simple values, or a heterogeneous, ordered collection of values, each of which can have any property, structural or constructed type.
 
-The `Map` data type is a heterogeneous, unordered collection of (_Key_, _Value_) pairs, where _Key_ is a literal and  _Value_ can have any property, structural, or composite type. 
+The `MAP` data type is a heterogeneous, unordered collection of (_Key_, _Value_) pairs, where _Key_ is a literal and  _Value_ can have any property, structural, or constructed type.
 
-Composite values can also contain `null`. 
+Constructed type values can also contain `null`.
 For more details, see xref::values-and-types/working-with-null.adoc[working with null].
 
+[[types-synonyms]]
+== Types and their synonyms
+
+The table below shows the types and their syntactic synonyms.
+These types (and the synonyms) can be used in the type predicate expressions and the property type constraints.
+However, not all types can be used in all places.
+// TODO: add link to type predicate expression section (once exists)
+// TODO: add link to property type constraint section (once exists)
+Some of these types are not yet supported in either type predicate expressions nor property type constraints.
+// TODO: Remove sentence about type not being supported once all types are allowed
+
+[.synonyms, opts="header", cols="2a,2a"]
+|===
+| Type | Synonyms
+| `BOOLEAN` | `BOOL`
+| `DATE` |
+| `DURATION` |
+| `FLOAT` |
+| `INTEGER` | `INT`, `SIGNED INTEGER`
+| `LIST<INNER_TYPE>` | `ARRAY<INNER_TYPE>`, `INNER_TYPE LIST`, `INNER_TYPE ARRAY`
```

Comment:
We might also allow `ANY<INTEGER | FLOAT> LIST` but not sure if I'd call that an inner type?

Reply:
We currently don't have anything around unions as they are not part of the already listed types in this chapter nor used in the constraints or expressions. Once they are the union type will be just another type that can be the inner type of a list, and probably also mention that if you use those variants of the list type you need the ANY<...> grouping.